ci/mock-dps-tests/mock-dps-provision.sh

```diff
@@ -0,0 +1,220 @@
+#!/bin/bash
+
+cd /src
+. ./ci/install-runtime-deps.sh
+. ./ci/mock-dps-tests/mock-dps-root-install.sh
+
+set -euo pipefail
+
+# Find the build output directory / the directory where CI extracted the artifact.
+#
+# For a local build, this would be target/x86_64-unknown-linux-gnu/debug.
+# For CI, this would be target/debug.
+cd "$(find target -type f -name mock-dps-server | head -n 1 | xargs dirname)"
+export LD_LIBRARY_PATH="${LD_LIBRARY_PATH:-}:$PWD"
+
+# Create directories needed for the tests.
+mkdir -p /run/aziot
+mkdir -p /etc/aziot/keyd/
+mkdir -p /var/lib/aziot/keyd
+mkdir -p /etc/aziot/certd
+mkdir -p /var/lib/aziot/certd
+mkdir -p /etc/aziot/identityd
+mkdir -p /var/lib/aziot/identityd
+
+uid=$(id -u)
+
+# Start mock-dps-server and wait for it to come up.
+./mock-dps-server --port 8443 --server-cert-chain "$SERVER_CERT_CHAIN" --server-key "$SERVER_KEY" &
+server_pid="$!"
+sleep 1
+
+# mock-dps-server does not authenticate, so tests can use any arbitrary credentials.
+echo 'mock-dps-provision' | base64 > device_id_symkey
+
+touch ~/.rnd
+>device_id_cert.conf cat <<-EOF
+[ device_id_cert ]
+basicConstraints = CA:FALSE
+keyUsage = digitalSignature, nonRepudiation, keyEncipherment
+extendedKeyUsage = clientAuth, emailProtection
+EOF
+
+openssl genrsa -out device_id_certkey.pem
+openssl req -new -key device_id_certkey.pem -subj "/CN=mock-dps-provision" -out device_id_certreq.pem
+
+openssl x509 -req \
+    -in device_id_certreq.pem \
+    -extfile device_id_cert.conf -extensions device_id_cert \
+    -signkey device_id_certkey.pem -sha256 \
+    -out device_id_cert.pem
+
+>/etc/aziot/keyd/config.toml cat <<-EOF
```

Comment:
Can you do these with `aziotctl config apply` instead? Would be nice to test it too.

Reply:
I tried that first, but aziotctl expects a package installation (it checks for the aziotks/aziotid/aziotcs users, uses systemctl restart, etc.). I figured this way was simpler.

Reply:
Ah, right.